src/core/index.ts

```diff
@@ -19,9 +29,165 @@ export class DevAssist {
 
 	private async startTask(task?: string): Promise<void> {
 		this.messages = [];
-		await this.providerRef.deref()?.postStateToWebview();
-
+		this.apiConversationHistory = [];
+		// TODO: At each prominent stage we need to show a message to the user as a update.
+		// await this.providerRef.deref()?.postStateToWebview(); // This should be called after each message update and handled in UI.
+		
 		await this.say("text", task);
+
+		await this.initiateTaskLoop([
+			{
+				type: "text",
+				text: `<task>\n${task}\n</task>`,
+			}
+		]);
+	}
+
+	private async initiateTaskLoop(userContent: any): Promise<void> {
+		let nextUserContent = userContent;
+		let includeFileDetails = true;
+		const didEndLoop = await this.recursivelyMakeClaudeRequests(nextUserContent, includeFileDetails);
+		includeFileDetails = false; // we only need file details the first time
+	}
+	
+
+	async recursivelyMakeClaudeRequests(
+		userContent: any,
+		includeFileDetails: boolean = false
+	): Promise<boolean> {
+		if (this.abort) {
+			throw new Error("DevAssist instance aborted")
+		}
+
+		await this.say(
+			"api_req_started",
+			JSON.stringify({
+				request:userContent
+			})
+		);
+
+		// TODO: FIXME: Environment is coming Undefined
+		// const environmentDetails = await this.loadContext(userContent, includeFileDetails)
+		const environmentDetails = "NONE"
+		// add environment details as its own text block, separate from tool results
+		userContent.push({ type: "text", text: environmentDetails })
+
+		await this.addToApiConversationHistory({ role: "user", content: userContent })
+
+		try {
+			
+			this.assistantMessageContent = []
+			this.didCompleteReadingStream = false
+			this.userMessageContent = []
+
+			const stream = this.attemptApiRequest(-1) // TODO -1 is a placeholder for now. For multiple communication we need to replace it by last message index
+			let assistantMessage = ""
+			try {
+				for await (const chunk of stream) {
+					assistantMessage += chunk.text
+					// TODO: parse raw assistant message into content blocks
+					// TODO: present content to user
+					console.log(assistantMessage);
+				}
+			} catch (error) {
+				this.abortTask() 
+			}
+
+			this.didCompleteReadingStream = true
```

Comment:
Are we planning on streaming the chunks directly to webview UI as and when the extension gets it from the llm or are we planning sending the whole message at once?

Reply:
I would need the data to be sent to the webview UI in JSON format. as discussed in #3

Reply:
Currently the data is coming in chunks that i am storing in a variable. @Sinduja98 is integrating parsing the each chunk to identify if it is a text or tool_use. For now current implementation we dont need to show data in the UI. 

Once all the data is rendered the content which is wrapped inside the <attempt_complesion> should be sown in UI. Which i will discuss with you.